### PR TITLE
Copier service

### PR DESCRIPTION
**Pull request by benbjohnson**
## Overview

This pull request adds the copier service which allows one server to copy shards from another server. This will be used for moving shards in the cluster.

## Additional Notes

- Utilitizes the existing multiplex'd TCP connection on the server.
- Protobuf headers are used for request & response to allow for future growth.
- Response body sent as `<uint64(length)><bytes...>`
- `copier.Client` provided for easier integration.
- `io.WriterTo` added to the `tsdb.Engine` interface and to `b1` & `bz1` implementations.

### CODE REVIEW

**Comment by otoolep** [comment]
Seems reasonable, nothing jumped out at me. +1

File: services/copier/service.go [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+		return fmt.Errorf("write request body: %s", err)
+	}
+
+	return nil
+}
+
+// readResponse reads and unmarshals a Response from r.
+func (c *Client) readResponse(r io.Reader) (*internal.Response, error) {
+	// Read response length.
+	var n uint32
```
Comment (jwilder) [inline services/copier/service.go]: Should this be a `uint64`?  Shards files can be larger than 4gb, right?

Comment (benbjohnson) [inline services/copier/service.go]: That's just for the response header. The shard writes out its data size as a `uint64`.

Comment (jwilder) [inline services/copier/service.go]: Oh.  Maybe this is just the length of the protobuf response.

**Pull request merged**